### DISCUSSION TITLE
Long-term sustainability?

### DISCUSSION THREAD

**Discussion by EcceMaterTua** [RyotaUshio/obsidian-pdf-plus#208] [Q&A] [answered]
First - WOW

This plugin is exactly what I've been waiting/hoping for in Obsidian. Much smoother than Zotero imports. Thank you for developing.

Regarding sustainability, I have one concern. When I disable the plugin, I'm still able to navigate from the Obsidian hyperlink to the PDF section (the green check in the picture), which is fantastic.

The problem is, after disabling, I lose the ability to navigate from PDF highlights back to the location in Obsidian (the red X), as well as lose the highlights in the PDF entirely. I've turned on the "Enable PDF editing" without any luck.

![2024-05-24 12_56_33_v2](https://github.com/RyotaUshio/obsidian-pdf-plus/assets/134634999/cf4768ca-9cc4-45f1-87f4-6d27a87d8184)

What happens if this plugin is no longer developed? Is there a way to keep PDF highlights and double-click those to navigate back to the Obsidian note?

**Comment by RyotaUshio** [comment]
Unfortunately, I think it's impossible to go back to the markdown note by double-clicking highlights WITHOUT PDF++, because it is purely PDF++'s feature.

Instead, you can use the backlink view for that purpose although it will not be as handy as double-click.

Donation is a good idea to help the constant maintenance and is very much appreciated, but it does not ensure the continuous development for, say, the next 10 years.
The good news is that this is an open-source project distributed under the most permissive MIT license, which means anyone around the world can fork it and take over the maintenance even if I ever stopped working on it.

**Reply by EcceMaterTua** [reply]
That makes sense, thanks for your fast reply.

With that being understood, could you help with a couple other things?

1. I don't know how all of this works, but could PDF++ at some point be "absorbed" into the native software by the Ob devs? This would be ideal I think, but if not, I'm still consoled that it's OS.
2. Even if the "PDF-to-Ob MD File" hyperlinking won't work after disabling PDF++, I think I'd be okay with that if the highlights would remain in the PDF. But I'm having trouble getting that to work. I have the below settings turned on:

![image](https://github.com/RyotaUshio/obsidian-pdf-plus/assets/134634999/4e4374a1-83cf-4d10-9009-89e26bfc02f8)

With those on, when I highlight the PDF, I see the highlights, but if I then disable PDF++, the highlights disappear in both the Ob file and the source file. Am I misunderstanding something? Or is there a different way to make the highlights permanent in either the source pdf file or Ob pdf file (or both?)?

Ultimately, I think I'd go forward using PDF++ if there was a way to have the highlights remain after disabling the plugin. Is there a way to do that?

As for a donation, I'm still in school for a couple years, so I'm unable to support at this time, but I think PDF++ is one of the best plugins available in Ob, so I'll very likely offer support when I graduate and have some income.

Thanks for developing such an incredible plugin!

**Reply by tprotopopescu** [reply]
To write the highlight to the pdf you have to also toggle 'write highlights to file by default' in the settings menu (under "Pdf annotations") or toggle it from the pdf toolbar (the disk icon).

**Reply by RyotaUshio** [reply, marked as answer]
@EcceMaterTua Sorry for the late reply.

1. At the end of the day, only the Obsidian team would know. But there have been several community plugins merged into the Obsidian core (like Admonition), so it's possible.
2. Everything @tprotopopescu said 🎉 (Thanks!)

**Reply by EcceMaterTua** [reply]
Thank you both. I see now the toggle on the PDF screen itself. I mistakenly thought the setting was the "toggle," but the way it's actually setup is much better. Thanks to you both!
